tests/ext/pynamodb/test_pynamodb.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from pynamodb.attributes import UnicodeAttribute
+from pynamodb.exceptions import VerboseClientError
+from pynamodb.models import Model
+
+from aws_xray_sdk.core import patch
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.context import Context
+
+patch(('pynamodb',))
+
+
+@pytest.fixture(autouse=True)
+def construct_ctx():
+    """
+    Clean up context storage on each test run and begin a segment
+    so that later subsegment can be attached. After each test run
+    it cleans up context storage again.
+    """
+    xray_recorder.configure(service='test', sampling=False, context=Context())
+    xray_recorder.clear_trace_entities()
+    xray_recorder.begin_segment('name')
+    yield
+    xray_recorder.clear_trace_entities()
+
+
+def test_exception():
```

Comment:
Could you add a test to guard the behavior where vendored requests only emit subsegments when talking to DynamoDB? In other words after patching `pynamodb` other AWS services are still captured only through original patched botocore code.

Reply:
I find it pretty difficult to write proper tests for the whole stuff, thanks to all the patching. I added a test now to check that only a single subsegment gets created when the pynamodb patch is enabled, but another AWS service is called. That single subsegment gets created, because the botocore patch is active in this case too.